### PR TITLE
refactor: enforce no runtime in pallet

### PR DESCRIPTION
**Pull request by weichweich**
## No Ticket

This PR enforces that no pallet* has dependencies to any runtime code. Which boils down to:

* All creates in `pallet/*` have no dependency to any crates in `nodes/` and `runtimes/*`
* `runtimes/*` has dependencies to `/pallets/*` but not to `/nodes*`
* `/nodes/*` has dependencies to everything.

Why would we do that!? It makes the test for each pallet more redundant!

### In a pallet: depending on `runtime-common` prevents us from having common configuration for that pallet.

If a pallet has a dependency to `runtime-common`, we can't have a dependency the other way. So if we need to have a configuration struct in a pallet, this struct can't get set inside the runtime common pallet. Or in more general if we have a custom type in a pallet, that depends on `runtime-common` we can't use this type in the `runtime-common` crate.

### Pallets should not depend on runtime code in general

A pallet should have no dependency on runtime specifics anyways.
Pallets are usable in what ever runtime you want.
Using runtime code in a pallet would mean that we bind the pallet to this specific runtime (even if it's only used in tests).

*I didn't touch the staking pallet since this is a bit more work and it wasn't needed for the pallet I was working on.

## Checklist:

- [x] I have verified that the code works
  - [x] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: pallets/kilt-launch/src/mock.rs [hunk 1]
```diff
@@ -125,16 +125,12 @@ impl kilt_launch::Config for Test {
...
-	pub const MinVestedTransfer: Balance = MIN_VESTED_TRANSFER_AMOUNT;
-}
-
 impl pallet_vesting::Config for Test {
 	type Event = Event;
 	type Currency = Balances;
 	type BlockNumberToBalance = ConvertInto;
 	// disable vested transfers by setting min amount to max balance
-	type MinVestedTransfer = MinVestedTransfer;
+	type MinVestedTransfer = runtime_common::constants::MinVestedTransfer;
```
Comment (ntn-x2) [inline pallets/kilt-launch/src/mock.rs:133]: I guess we want to remove this dependency here?

Comment (weichweich) [reply pallets/kilt-launch/src/mock.rs:133]: yeah I skipped a few pallets because they where too much work. This seems to be one of them. 🙈

**Review by ntn-x2 — COMMENTED** [review]
As mentioned, there's still a dependency in the lock file from the launch and the staking pallets towards the `runtime-common` crate. There are also few other comments, but the cleanup is very nice to have!

File: runtimes/common/src/constants.rs [hunk 2]
```diff
@@ -272,6 +376,15 @@ pub mod web3_names {
...
 	pub const MAX_NAME_BYTE_LENGTH: u32 = 121;
 	pub const DEPOSIT: Balance = deposit(2, MAX_NAME_BYTE_LENGTH);
+
+	parameter_types! {
+		pub const Web3NameDeposit: Balance = DEPOSIT;
+		pub const MinNameLength: u32 = MIN_LENGTH;
+		pub const MaxNameLength: u32 = MAX_LENGTH;
+	}
+
+	// pub type Web3Name<R> = pallet_web3_names::web3_name::AsciiWeb3Name<R,
```
[the remark this answers is not in the document]
Comment (weichweich) [reply runtimes/common/src/constants.rs]: Ah yeah. A left over from the other PR. [6728443](https://github.com/KILTprotocol/mashnet-node/pull/349/commits/67284439db134f1020d33b18679f428b2bb79c68)

File: pallets/did/src/mock.rs [hunk 3]
```diff
@@ -387,14 +385,15 @@ impl DeriveDidCallAuthorizationVerificationKeyRelationship for Call {
 	// Always return a System::remark() extrinsic call
 	#[cfg(feature = "runtime-benchmarks")]
 	fn get_call_for_did_call_benchmark() -> Self {
+		use sp_std::vec;
```
[the remark this answers is not in the document]
Comment (weichweich) [reply pallets/did/src/mock.rs]: [369d3ff](https://github.com/KILTprotocol/mashnet-node/pull/349/commits/369d3ff31fb785b8c6ddca8e50e3ee78f05a5c03) It does indeed work!

File: pallets/kilt-launch/src/mock.rs (hunk 1 above)
Comment (weichweich) [reply pallets/kilt-launch/src/mock.rs]: [0007b8f](https://github.com/KILTprotocol/mashnet-node/pull/349/commits/0007b8fe13216a1e6bcddf82ff80944a5be0675e) & [4b14edb](https://github.com/KILTprotocol/mashnet-node/pull/349/commits/4b14edbdcb75bb032b59804f8b5a748eccd38c4e)
Kilt Launch & parachain staking should be the only two missing.

File: pallets/did/src/mock.rs [hunk 4]
```diff
@@ -413,30 +412,30 @@ pub fn generate_test_did_call(
...
-	deleted_dids: Vec<TestDidIdentifier>,
-	ctypes_stored: Vec<(TestCtypeHash, TestCtypeOwner)>,
+	dids_stored: Vec<(DidIdentifier, DidDetails<Test>)>,
+	service_endpoints: Vec<(DidIdentifier, Vec<DidEndpoint<Test>>)>,
+	deleted_dids: Vec<DidIdentifier>,
+	ctypes_stored: Vec<(CtypeHash, DidIdentifier)>,
 	balances: Vec<(AccountIdOf<Test>, Balance)>,
 }
 
 impl ExtBuilder {
```
[the remark this answers is not in the document]
Comment (ntn-x2) [reply pallets/did/src/mock.rs:429]: I realised that the impl block is already in a module that is used only for mocks and tests, so my comment does not really make sense. Please ignore it.

File: pallets/did/src/did_details.rs [hunk 5]
```diff
@@ -177,7 +177,7 @@ pub trait DidVerifiableIdentifier {
 	) -> Result<DidVerificationKey, SignatureError>;
 }
 
-impl DidVerifiableIdentifier for runtime_common::DidIdentifier {
+impl<I: AsRef<[u8; 32]>> DidVerifiableIdentifier for I {
```
[the remark this answers is not in the document]
Comment (weichweich) [reply pallets/did/src/did_details.rs:180]: For the implementation the requirement `I: AsRef<[u8; 32]>` should be enough. But I would like to limit it to the actual type that is the `DidIdentifier` and not just all things that can be referenced as a `[u8; 32]`. Which is also an `AccountId`. But I think there is no way around that. 🤷 Adding more constraints maybe reduce the number of types this is implemented for, but I'm not sure if it's worth it? 🤔

[the remark this answers is not in the document]
Comment (ntn-x2) [reply pallets/did/src/did_details.rs:180]: I would not spend too much on this, but if that's the only trait required right now in the method implementation and it's compiling, I would go on with that.

**Pull request merged**